### ISSUE TITLE
Faceted search show the price of the default combination instead of the selected combination

### ISSUE THREAD

**Issue by marionf** [PrestaShop/PrestaShop#15164]
Creates two products with combinations S-M-L
Apply for these 2 products for combination M an impact on the price
In front-office select combination M with faceted search
You should see the price of the selected combination

---
#### Describe the bug

When using faceted search we got a problem - filters shows wrong results in products with combinations (especially prices). It shows only price of default combination, not the right one which fulfill the filter requirements.

#### Expected behavior

Output from filter, should be only the combination and it's prices which is expected by a customer (which fill right filter attributes)

#### Steps to Reproduce

1. Create product with 3 combinations 
2. Each combination has different price
3. Go to any page with products
4. Filter (look at price slider - its max value is max price of default combination)
5. In filter choose Attribute which is not default
6. Look - it shows the right product but the price is from default combination (which don't meet filter attribute value).

**Screenshots**
https://drive.google.com/file/d/1tBOxqabHX2ERdCvLkEe4WZlYpqAxTvkV/view

#### Additional information

To sum up: If you got X product with sizes S, M, L which has different price then you got a problem. If customers filter by size for example L, they will se WRONG: price range (slider), prices on products (they will see S prices).

* PrestaShop version: 1.7.7.8
* PHP version: 7.3

**Comment by TheKruspe** [comment]
I was able to fix this by adding four lines of code within the `ps_facetedsearch` repo.

In `src/Filters/Products.php`, `getProductByFilters`, I added

    // this block
    if (isset($selectedFilters['id_attribute_group'])) {
        $this->searchAdapter->addSelectField('id_product_attribute');
    }
    // end block
    $matchingProductList = $this->searchAdapter->execute();

In `src/Adapter/MySQL.php`, `useFiltersAsInitialPopulation`, I added

    $this->setSelectFields(
            [
                'id_product',
                'id_product_attribute', // this line
                'id_manufacturer',
                'quantity',
                'condition',
                'weight',
                'price',
                'sales',
            ]
        );
        
This also fixes https://github.com/PrestaShop/PrestaShop/issues/15738.

**Comment by Xillver88** [comment]
Hi,

i have tested the solution and it works! So thank you very much, just 1 question? When will this be officially released as an updated to facetedSearch module?

**Comment by Hlavtox** [comment]
Fixed with Prestashop 9 and last versions of faceted search. Also displays proper image etc.
